src/tools/rbd_cache/ObjectCacheStore.cc

```diff
@@ -0,0 +1,152 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "ObjectCacheStore.h"
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rbd_cache
+#undef dout_prefix
+#define dout_prefix *_dout << "rbd::cache::ObjectCacheStore: " << this << " " \
+                           << __func__ << ": "
+
+namespace rbd {
+namespace cache {
+
+ObjectCacheStore::ObjectCacheStore(CephContext *cct, ContextWQ* work_queue)
+      : m_cct(cct), m_work_queue(work_queue),
+        m_cache_table_lock("rbd::cache::ObjectCacheStore"),
+        m_rados(new librados::Rados()) {
+}
+
+ObjectCacheStore::~ObjectCacheStore() {
+
+}
+
+int ObjectCacheStore::init(bool reset) {
+
+  int ret = m_rados->init_with_context(m_cct);
+  if(ret < 0) {
+    lderr(m_cct) << "fail to init Ceph context" << dendl;
+    return ret;
+  }
+
+  ret = m_rados->connect();
+  if(ret < 0 ) {
+    lderr(m_cct) << "fail to conect to cluster" << dendl;
+    return ret;
+  }
+  //TODO(): check existing cache objects
+  return ret;
+}
+
+int ObjectCacheStore::do_promote(std::string pool_name, std::string object_name) {
```

Comment:
Looks like you still need a cache policy for handling the max size of your cache and detecting hot/cold objects worth promoting.

Reply:
Yes, there's some pending code on policy based promote/evict. Will push out for  review in next commit.